### PR TITLE
Create separate queues for preferences records

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Create queues per device per categories.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
Marked as draft because does not have updated tests and delete queues code does not removes queues for prefs.

**Comment by AlexeyBarabash** [comment]
Pushed fix for error found while preparing brave-core corresponding PR.

File: client/requestUtil.js [hunk 1]
```diff
@@ -224,9 +230,9 @@ RequestUtil.prototype.shouldListObject = function (startAt, category) {
   let startAtToCheck = this.normalizeTimestampToMs(startAt, currentTime)
 
   return !startAtToCheck ||
-      (currentTime - startAtToCheck) > parseInt(s3Helper.SQS_RETENTION, 10) * 1000 ||
-      category !== proto.categories.BOOKMARKS ||
-      this.listInProgress === true
+       (currentTime - startAtToCheck) > parseInt(s3Helper.SQS_RETENTION, 10) * 1000 ||
```
Comment (darkdh) [inline client/requestUtil.js]: nit: weird indention

Comment (AlexeyBarabash) [reply client/requestUtil.js]: fixed that in https://github.com/brave/sync/pull/331/files#diff-93951cfb12e03132f0bd5b9a600a0a27R234

File: client/requestUtil.js [hunk 2]
```diff
@@ -196,7 +197,7 @@ RequestUtil.prototype.list = function (category, startAt, maxRecords, nextContin
     }
     // We poll from SQS
     let notificationParams = {
-      QueueUrl: `${this.SQSUrl}`,
+      QueueUrl: `${this.SQSUrlByCat[category]}`,
```
Comment (SergeyZhukovsky) [inline client/requestUtil.js]: I think we should add a check in that place does `this.SQSUrlByCat` contains a `category` item or at least if it's size is not 0. And put that info in the log to easy find if something is wrong.

Comment (AlexeyBarabash) [reply client/requestUtil.js]: Pushed the fix according to review

**Pull request merged**